quickwit/quickwit-cli/src/lib.rs

```diff
@@ -256,23 +257,18 @@ fn get_resolvers(
 /// Optionally, it takes a `SourceConfig` that will be checked instead
 /// of the index's sources.
 pub async fn run_index_checklist(
-    metastore: &dyn Metastore,
+    metastore: &mut MetastoreServiceClient,
     storage_resolver: &StorageResolver,
     index_id: &str,
     source_config_opt: Option<&SourceConfig>,
 ) -> anyhow::Result<()> {
     let mut checks: Vec<(&str, anyhow::Result<()>)> = Vec::new();
 
-    // The metastore is file-backed, so we must check the storage first.
```

Comment:
This check is useless. The check connectivity of the metastore will do the right check.

Reply:
Leave it. The checklist will tell you it's a metastore issue, whereas it stems from the storage.

Reply:
Ok, will change the comment then.

Reply:
I would prefer to do the right check in the metastore method and make sure the error displayed is clean.